OloEditor/src/Panels/SceneHierarchyPanel.cpp

```
@@ -614,13 +995,65 @@ namespace OloEngine
+                if (s_DrawComponentCmdHistory && s_DrawComponentScene)
+                {
+                    struct EditState
+                    {
+                        bool isEditing = false;
+                        T snapshot{};
+                    };
+                    static std::unordered_map<u64, EditState> s_editStates;
+                    auto& editState = s_editStates[static_cast<u64>(entity.GetUUID())];
+
+                    // Maintain a stable snapshot of the component before any edit session
+                    if (!editState.isEditing)
```

Clang (14.0.6)

[note] +4, including nesting penalty of 3, nesting level increased to 4

(clang)